Display subbasin results view via water quality tab link

kellyi opened the pull request:
## Overview

This PR adds a link to MapShed water quality results for valid WKAOIs to kick off subbasin modeling, along with a new panel which will be used to display subbasin results in #2650.

In this PR we don't add the subbasin results to the new scenario/results/collection, but just log to the console that it's succeeded. However, we do keep the old mapshed results around to afford not having them replaced by the incoming subbasin results.

Connects #2647 

## Demo

![screen shot 2018-04-02 at 6 33 50 pm](https://user-images.githubusercontent.com/4165523/38219623-6ecdd29a-36a4-11e8-8d50-d736c06f6395.png)

![screen shot 2018-04-02 at 6 33 58 pm](https://user-images.githubusercontent.com/4165523/38219626-727c94ee-36a4-11e8-95ba-d4d4686b9621.png)

... and meanwhile, in the console ...

![screen shot 2018-04-02 at 6 27 55 pm](https://user-images.githubusercontent.com/4165523/38219553-1e174868-36a4-11e8-9b6e-a1ff0c2a16b4.png)

## Testing
- get branch, rebuild, etc
- try modeling a non-subbasin-valid-aoi and verify that you don't see the new link
- model a HUC 8 or HUC 10 and verify that you do see the new link, then check that clicking it shows the new view and also commences subbasin modeling, which will be logged to the console and show the correct result from its last network request

arottersman commented:
Testing this out now!

kellyi commented on src/mmw/js/src/modeling/views.js:
```diff
@@ -1032,24 +1034,54 @@ var ResultsDetailsView = Marionette.LayoutView.extend({
+
+        this.subbasinRegion.show(new gwlfeSubbasinViews.ResultView({
+            hideSubbasinHotSpotView: this.hideSubbasinHotSpotView,
+            model: this.collection.findWhere({ name: 'subbasin' }),
```
Good catch -- I removed the mock subbasin model (the stray "remove this" comment was supposed to go with it) since we currently don't persist the results.

kellyi commented on src/mmw/js/src/modeling/views.js:
```diff
@@ -1032,24 +1034,54 @@ var ResultsDetailsView = Marionette.LayoutView.extend({
+
+    showSubbasinHotSpotView: function() {
+        this.panelsRegion.empty();
+        this.contentRegion.empty();
```
:+1: Sounds good. I'll adjust this.

kellyi commented:
Pushed fixups to address most of the comments above. Still to come is starting on the water quality tab on clicking back from subbasin. As best I can tell the state of what tab we're on isn't really tracked anywhere except by adjusting an `.active` class from tab to tab and li to li in the UI? Are we storing it somewhere else?

arottersman commented:
> Are we storing it somewhere else?

I don't think so. The tabs all manage their own business. Perhaps it is finally time for it to be a modal?

kellyi commented:
Made an adjustment to only set the polling state true if we're in a non-subbasin task model. Also squashed and rebased on develop. @arottersman and I discussed designing the subbasin task/scenario/results models as part of a revised and larger version of #2650.

kellyi commented:
Thanks for your help with this!
